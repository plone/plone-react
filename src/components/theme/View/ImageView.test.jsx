import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';

import ImageView from './ImageView';

const mockStore = configureStore();

test('renders an image view component', () => {
  const store = mockStore({
    intl: {
      locale: 'en',
      messages: {},
    },
  });
  const component = renderer.create(
    <Provider store={store}>
      <ImageView
        content={{
          title: 'Hello World!',
          description: 'Hi',
          image: {
            download: 'http://localhost:8080/Plone/test-images/@@images/image',
            size: 123123,
            width: 1920,
            scales: {
              icon: {
                download:
                  'http://localhost:8080/Plone/test-images/@@images/image/icon',
                width: 32,
              },
              large: {
                download:
                  'http://localhost:8080/Plone/test-images/@@images/image/large',
                width: 768,
              },
              listing: {
                download:
                  'http://localhost:8080/Plone/test-images/@@images/image/listing',
                width: 16,
              },
              mini: {
                download:
                  'http://localhost:8080/Plone/test-images/@@images/image/mini',
                width: 200,
              },
              preview: {
                download:
                  'http://localhost:8080/Plone/test-images/@@images/image/preview',
                width: 400,
              },
              thumb: {
                download:
                  'http://localhost:8080/Plone/test-images/@@images/image/thumb',
                width: 128,
              },
              tile: {
                download:
                  'http://localhost:8080/Plone/test-images/@@images/image/tile',
                width: 64,
              },
            },
          },
        }}
      />
    </Provider>,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
