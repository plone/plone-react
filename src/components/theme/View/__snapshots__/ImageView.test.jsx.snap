// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an image view component 1`] = `
<div
  className="ui container view-wrapper"
>
  <h1
    className="documentFirstHeading"
  >
    Hello World!
  </h1>
  <p
    className="documentDescription"
  >
    Hi
  </p>
  <a
    href="/test-images/@@images/image"
  >
    <picture>
      <img
        alt="Hello World!"
        loading="lazy"
        role="img"
        size="preview"
        src="http://localhost:8080/Plone/test-images/@@images/image/listing"
        style={
          Object {
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
    </picture>
    <noscript
      dangerouslySetInnerHTML={
        Object {
          "__html": "
              <img
                src=\\"http://localhost:8080/Plone/test-images/@@images/image/listing\\"
                srcset=\\"/test-images/@@images/image/listing 16w, /test-images/@@images/image/icon 32w, /test-images/@@images/image/tile 64w, /test-images/@@images/image/thumb 128w, /test-images/@@images/image/mini 200w, /test-images/@@images/image/preview 400w, /test-images/@@images/image/large 768w, /test-images/@@images/image 1920w\\"
                alt=\\"Hello World!\\"
                className=\\"undefined\\"
                role=\\"img\\"
                loading=\\"lazy\\"
                style={{ width: '100%', objectFit: 'cover' }}
            ",
        }
      }
    />
    <figcaption>
      Size: 123 kB
        —  
      Click to download full sized image
    </figcaption>
  </a>
</div>
`;
